internal/models/ingrediente.go

```diff
@@ -0,0 +1,17 @@
+package models
+
+import "time"
+
+type Ingrediente struct {
+	producto       Producto
+	cantidad       float32
+	fechaCaducidad *time.Time
+}
+
+func NewIngrediente(producto Producto, cantidad float32, fechaCaducidad *time.Time) Ingrediente {
```

Comment:
Vamos a ver si entendemos lo que es crear un modelo. Ya he dicho que no es poner nombre a las cosas. También es tener una única fuente de verdad. Dado que cualquiera puede crear un `Producto` y añadirlo aquí con tipos diferentes, *sigue sin haber una única fuente de verdad*. Pero el que no lo haya revela dos problemas un poco más profundos.

1. ¿Dónde están los productos que vas a usar para añadir aquí? Si yo pillo y creo un producto por las buenas desde donde quiera que llamo al ingrediente, ¿qué me lo impide hacerlo? Crear un modelo es hacer una *entidad* que sepa *gestionar* cosas como los `Producto`s creando, por tanto, un modelo del problema.
2. ¿Cómo te refieres a este ingrediente? Estamos de acuerdo que es un objeto valor, pero suponte que tienes que comprobar si un ingrediente determinado se ha añadido ya o no. ¿Como lo haces? Tienes dos niveles de indirección: primero acceder al objeto y segundo acceder al *nombre* del objeto

Ya os dije varias veces que una de las cosas que tenéis que intentar hacer es ver *cómo* se va a usar esto. Es decir, hacer el *user journey* pero desde el punto de vista del programador. ¿Cómo se hace? ¿Es fácil hacerlo? Si no lo es, es que el modelo es incorrecto.

Reply:
Corregido en estos commits:
- (https://github.com/juanbarearojo/privateChef/pull/15/commits/8614973b48f1ed64460f8151e2f93856f40d0b68)
- (https://github.com/juanbarearojo/privateChef/pull/15/commits/28a7454e8942b2fc5767212f8fc5bfe14095553e)